engine/servermaster/jobmanager.go

```diff
@@ -177,9 +184,64 @@ func (jm *JobManagerImpl) GetJob(ctx context.Context, req *pb.GetJobRequest) (*p
 		return nil, err
 	}
 
+	// if job status is running, forward the request to jobmaster openapi
+	if masterMeta.StatusCode == frameModel.MasterStatusInit {
+		status, err := getJobDetailFromJobMaster(ctx, masterMeta.Addr, req.Id)
+		return buildPBJobWithStatus(masterMeta, status, err)
+	}
+
 	return buildPBJob(masterMeta)
 }
 
+// TODO: add a timeout here
+func getJobDetailFromJobMaster(ctx context.Context, address string, jobID string) (*pb.Status, error) {
+	queryJobPath := fmt.Sprintf(jobStatusAPIFormat, jobID)
+	// TODO: TLS?
+	cli, err := httputil.NewClient(nil)
+	if err != nil {
+		return nil, derrors.ErrJobManagerNewHTTPClientFail.Wrap(err)
+	}
+	url := fmt.Sprintf("http://%s%s", address, queryJobPath)
+	log.Info("get job detail from job master", zap.String("url", url))
+	resp, err := cli.Get(ctx, url)
+	if err != nil {
+		return nil, derrors.ErrJobManagerGetJobDetailFail.Wrap(err)
+	}
```

Comment:
Can this logic be encapsulated inside some client object?

Reply:
Ok.